mycore-base/src/main/java/org/mycore/common/xml/MCRXMLHelper.java

```diff
@@ -130,6 +134,30 @@ public static void validate(Document doc, String schemaURI) throws SAXException,
         validator.validate(new JDOMSource(doc));
     }
 
+    /**
+     * Loads a XSD as source from an URI. If the uri has http or https scheme, the catalog.xml will be tried first,
+     * otherwise the URI resolver.
+     *
+     * @param schemaURI uri to the XSD
+     * @return a resolved xsd as source
+     * @throws IOException
+     * @throws TransformerException
+     */
+    public static Source getXSDSource(String schemaURI) throws IOException, TransformerException {
+        String url = schemaURI;
+        if (schemaURI.startsWith("http://") || schemaURI.startsWith("https://")) {
+            InputSource entity = MCREntityResolver.instance().resolveEntity(null, url);
```

Comment:
Code should not depend on specific scheme as any URI could be mapped in XMLCatalog